src/main/java/gov/cabinetoffice/gap/adminbackend/services/GrantAdvertService.java

```diff
@@ -320,7 +326,7 @@ public GrantAdvert publishAdvert(UUID advertId) {
 
         if (Boolean.FALSE.equals(contentfulAdvert.isPublished())) {
             final CMAEntry publishedAdvert = contentfulManagementClient.entries().publish(contentfulAdvert);
-            openSearchService.indexEntry(publishedAdvert);
+            sendMessageToQueue(new SendAdvertToSqsDto(publishedAdvert.getId(), "ADD"));
```

Comment:
I think this function should be the very last thing this endpoint does - as if we fail to save to the DB for whatever reason, this endpoint will be hit multiple times due to the publish advert lambdas retry behaviour, which will result in us sending multiple messages to our new publish elastic lambda. I think the end state would then be duplicate adverts showing up in elastic in this edge case

Reply:
Jamie and I tested elastic last week, it's idempotent so not sure there would be duplicates but the lambda on the other end of this exchange might end up with errors from contentful so good shout nonetheless.